pkg/migration/generate.go

```diff
@@ -830,6 +835,45 @@ func (h *Handler) addMonitoringSpec(cluster *corev1.StorageCluster, ds *appsv1.D
 	return nil
 }
 
+func (h *Handler) addCustomAnnotations(cluster *corev1.StorageCluster, ds *appsv1.DaemonSet) error {
+	podList := &v1.PodList{}
+	err := h.client.List(
+		context.TODO(),
+		podList,
+		&client.ListOptions{
+			Namespace:     ds.Namespace,
+			LabelSelector: labels.SelectorFromSet(map[string]string{"name": "portworx"}),
+		},
+	)
+	if err != nil {
+		return err
+	}
+
+	customAnnotations := make(map[string]string)
+	for _, pod := range podList.Items {
+		owner := metav1.GetControllerOf(&pod)
+		if owner != nil && owner.UID == ds.UID && pod.Annotations != nil {
+			for k, v := range pod.Annotations {
+				if k != constants.AnnotationPodSafeToEvict {
```

Comment:
Almost all spec generator daemonsets have this annotation - `portworx.com/install-source`. Let's skip that too.